### ISSUE TITLE
_mixture_ should be able to return List[Tuple[float, Operation]] instead of List[Tuple[float, matrix]]

### ISSUE THREAD

**Issue by Strilanc** [quantumlib/Cirq#1657]
Where it is required that the operations be unitary.

This is beneficial because it allows you to return mixtures of large operations, without having to generate their unitaries.

Fixing this mostly involves finding usages of mixture and making them account for this.

**Comment by dabacon** [comment]
You mean Gate instead of Operation?

**Comment by Strilanc** [comment]
Good point. A gate `_mixture_` would return gates, whereas an operation `_mixture_` would return operations. Then, in `GateOperation`, there would be a conversion.

**Comment by c-poole** [comment]
Is this a duplicate of #1626? Also, I can give this a shot.

**Comment by Strilanc** [comment]
Yup, it's a duplicate. Both by me. Four days apart. Wow.
